include/zephyr/drivers/sensor.h

```diff
@@ -599,6 +761,164 @@ static inline int z_impl_sensor_channel_get(const struct device *dev,
+	int8_t shift;
+	enum sensor_channel channel;
+};
+
+/*
+ * Generic data structure used for encoding the sample timestamp and number of channels sampled.
+ */
+struct __attribute__((__packed__)) sensor_data_generic_header {
+	uint64_t timestamp_ns;
+	size_t num_channels;
+	struct sensor_data_generic_channel channel_info[0];
+};
+
+/**
+ * @brief checks if a given channel is a 3-axis channel
+ *
+ * @param[in] chan The channel to check
+ * @return true if @p chan is :c:enum:`SENSOR_CHAN_ACCEL_XYZ`
+ * @return true if @p chan is :c:enum:`SENSOR_CHAN_GYRO_XYZ`
+ * @return true if @p chan is :c:enum:`SENSOR_CHAN_MAGN_XYZ`
+ * @return false otherwise
+ */
+#define SENSOR_CHANNEL_3_AXIS(chan)                                                                \
+	((chan) == SENSOR_CHAN_ACCEL_XYZ || (chan) == SENSOR_CHAN_GYRO_XYZ ||                      \
+	 (chan) == SENSOR_CHAN_MAGN_XYZ)
+
+/**
+ * @brief Get the sensor's decoder API
+ *
+ * @param[in] dev The sensor device
+ * @param[in] decoder Pointer to the decoder which will be set upon success
+ * @return 0 on success
+ * @return < 0 on error
+ */
+__syscall int sensor_get_decoder(const struct device *dev,
+				 const struct sensor_decoder_api **decoder);
+
+static inline int z_impl_sensor_get_decoder(const struct device *dev,
+					    const struct sensor_decoder_api **decoder)
+{
+	const struct sensor_driver_api *api = (const struct sensor_driver_api *)dev->api;
+
+	__ASSERT_NO_MSG(api != NULL);
+
+	if (api->get_decoder == NULL) {
+		*decoder = &__sensor_default_decoder;
+		return 0;
+	}
+
+	return api->get_decoder(dev, decoder);
+}
+
+/**
+ * @brief Reconfigure a reading iodev
+ *
+ * Allows a reconfiguration of the iodev associated with reading a sample from a sensor.
+ *
+ * <b>Important</b>: If the iodev is currently servicing a read operation, the data will likely be
+ * invalid. Please be sure the flush or wait for all pending operations to complete before using the
+ * data after a configuration change.
+ *
+ * It is also important that the `data` field of the iodev is a :c:struct:`sensor_read_config`.
+ *
+ * @param[in] iodev The iodev to reconfigure
+ * @param[in] sensor The sensor to read from
+ * @param[in] channels One or more channels to read
+ * @param[in] num_channels The number of channels in @p channels
+ * @return 0 on success
+ * @return < 0 on error
+ */
+__syscall int sensor_reconfigure_read_iodev(struct rtio_iodev *iodev, const struct device *sensor,
+					    const enum sensor_channel *channels,
+					    size_t num_channels);
+
+static inline int z_impl_sensor_reconfigure_read_iodev(struct rtio_iodev *iodev,
+						       const struct device *sensor,
+						       const enum sensor_channel *channels,
+						       size_t num_channels)
+{
+	struct sensor_read_config *cfg = (struct sensor_read_config *)iodev->data;
+
+	if (cfg->max < num_channels) {
+		return -ENOMEM;
+	}
+
+	cfg->sensor = sensor;
+	memcpy(cfg->channels, channels, num_channels * sizeof(enum sensor_channel));
```

Comment:
I see lots of special handling for 3-axis sensor types.  However the `decode` routine appears to modify the channel type from the 3-axis identifier to a single axis identifier.

Can you eliminate most of the 3-axis logic in default_rtio_sensor  if you just convert the single 3-axis type to 3 single axis types here?

```
count  = 0;
for (i = 0; i < num_channels; i++, channels++)
    if (SENSOR_CHANNEL_3_AXIS(*channels))
        cfg->channels[count++] = *channels - 3;
        cfg->channels[count++] = *channels - 2;
        cfg->channels[count++] = *channels - 1;
    else
        cfg->channels[count++] = *channels;
cfg->count = count;
```

Reply:
That was my original implementation actually. Doing things this way allowed me to compress the data quite a bit more. With the current implementation getting the 7 channels from an accel/gyro (3 accel, 3 gyro, and die temp) actually uses less memory than the existing fetch/get API. It's being a bit picky with bytes so I'm happy to go back to the simpler implementation.

Reply:
The logic to manage the 3-axis data as a single channel is difficult to parse and is distributed througout the default_rtio_sensor code, so I'm in favor of the simpler implementation.
Although, this will likely require some fun macrobatics with `SENSOR_DT_READ_IODEV` to detect the 3 axis accelerometer types.

Reply:
Done